### ISSUE TITLE
"no-used-vars" rule incompatible with JSDoc

### ISSUE THREAD

**Issue by Spy-Seth** [eslint/eslint#2813]
If I use the `no-used-vars` rule on the following file: 

```javascript
'use strict';

var Bar = require('src/Bar');

/**
 * @param {Bar} bar
 * @constructor
 */
var Foo = function Foo(bar) {
    this.baz = bar;
    //...
};

module.exports = Foo;
```

It will trigger this warning:
```
$ eslint foo.js                                                                                                                                    
3:4  error  Bar is defined but never used  no-unused-vars

✖ 1 problem (1 error, 0 warnings)
```

I do not use the `Bar` variable in my code. But I did use it in the JSDoc. This really usefull in big with lot of Object with the same name in different namespace/context. It helps to understand the code.

Is possible to add an option to the `no-used-vars` to manage reference to a variable in JSDoc?

**Comment by nzakas** [comment]
No, JSDoc can't be used as a "use" of a variable. Your best bet is to use a comment to disable the rule for that line: http://eslint.org/docs/user-guide/configuring#configuring-rules

**Comment by hzoo** [comment]
An alternative I guess is to not use jsdoc for params and use flow (or typescript?) - `babel-eslint` will lint flow for `no-unused-vars/no-undef` now

**Comment by artin-phares** [comment]
From v2.3 typescript can check types in `.js` files and gather type definitions from `jsdoc` annotations.  

It would be very useful to have `jsdoc` as valid 'use' of a variables.
